### PR TITLE
implement GUIDED_HEADING_TYPE default mavlink message

### PR DESCRIPTION
**Pull request by timtuxworth**
The MAVLink GUIDED_CHANGE_HEADING command will switch from the default to either HEADING_TYPE_HEADING or HEADING_TYPE_COG, but did not have a way to switch back to "default" this adds the code to implement HEADING_TYPE_DEFAULT which will switch the ArduPilot internal plane.guided_state.target_heading_type back to GUIDED_HEADING_NONE, which is the default value with it's own behavior.

Follow up to the MAVLink PR https://github.com/ArduPilot/mavlink/pull/372 which adds the implementation.

This PR also updates the mavlink submodule reference.

### CODE REVIEW

**Comment by timtuxworth** [comment]
Follow up to the MAVLink PR https://github.com/ArduPilot/mavlink/pull/372 which adds the implementation. How does this look @peterbarker , @IamPete1 ?

File: ArduPlane/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -973,22 +976,28 @@ MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_guided_slew_commands(const mavl
...
+        default:
             //  MAV_RESULT_DENIED  means Command is invalid (is supported but has invalid parameters).
             return MAV_RESULT_DENIED;
         }
 
         plane.g2.guidedHeading.reset_I();
 
         plane.guided_state.target_heading = new_target_heading;
-        plane.guided_state.target_heading_accel_limit = MAX(packet.param3, 0.05f);
+        plane.guided_state.target_heading_accel_limit = MAX(is_zero(packet.param3)? 10.0f : packet.param3 , 10.0f); // the, previous limit of 0.05 was 0.29 degrees, not very useful
```
Comment (IamPete1) [inline ArduPlane/GCS_Mavlink.cpp:1000]: Why this change?

Comment (timtuxworth) [reply ArduPlane/GCS_Mavlink.cpp:1000]: As per the comment @iampete // the, previous limit of 0.05 was 0.29 degrees, not very useful
The value is in radians so It was throttling the rate at which heading would change to very, very tiny increments that prevented the follow from  being able to track the target plane.

File: ArduPlane/GCS_Mavlink.cpp [hunk 2]
```diff
@@ -973,22 +976,28 @@ MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_guided_slew_commands(const mavl
 
         float new_target_heading = radians(wrap_180(packet.param2));
 
-        // course over ground
-        if ( int(packet.param1) == HEADING_TYPE_COURSE_OVER_GROUND) { // compare as nearest int
+        switch(HEADING_TYPE(packet.param1)) {
+        case HEADING_TYPE_COURSE_OVER_GROUND:
+            // course over ground
```
Comment (Ryanf55) [inline ArduPlane/GCS_Mavlink.cpp:981]: I think it's safe to remove these comments, it's obvious from the case what this is for.

File: ArduPlane/GCS_Mavlink.cpp (hunk 1 above)
Comment (IamPete1) [reply ArduPlane/GCS_Mavlink.cpp:1000]: It's a max not a min, it's saying the lowest it can be is  0.05 but it can be larger.

Comment (timtuxworth) [reply ArduPlane/GCS_Mavlink.cpp:1000]: > It's a max not a min, it's saying the lowest it can be is 0.05 but it can be larger.

Read that again - does it make sense to you?

Comment (IamPete1) [reply ArduPlane/GCS_Mavlink.cpp:1000]: > Read that again - does it make sense to you?

Yes, but then I'm dyslexic. Let me try again. 

Its a `MAX` operation. The output can be anything larger than 0.05.

Your complaint seems to be that 0.05 is too small? The `MAX` does not prevent you using a larger value.

File: ArduPlane/GCS_Mavlink.cpp (hunk 2 above)
Comment (timtuxworth) [reply ArduPlane/GCS_Mavlink.cpp:981]: I would but reviewers tend to not to like those kind of changes

**Review by IamPete1 — APPROVED** [review]
I think this looks good.

**Review by tridge — APPROVED** [review]
